### ISSUE TITLE
Training on half precision leads to nan values

### ISSUE THREAD

**Issue by isamu-isozaki** [afiaka87/glide-finetune#5]
I was training my model and I noticed that after just the first iteration I was running into nan values. As it turns out my gradients and input values/images were all normal but the adam optimizer by pytorch does has some weird behavior on float16 precision where it produces nans probably because of a divide by 0 error. A discussion can be found below

https://discuss.pytorch.org/t/adam-half-precision-nans/1765/4

I hear changing the epison parameter for the adam weights parameter when on half precisions works but I haven't tested it yet. Will make one once I tested.

And also let me say thanks for this repo. I wanted to fine tune the glide model and this made it so much easier.

**Comment by isamu-isozaki** [comment]
Made pr.
